Hw 34 solution

ipantazi opened the pull request:
Added `Book` entity, `DataProcessingException` class, `BookRepository` interface with methods and `BookRepositoryImpl` class.
Added `BookService` interface with methods and `BookServiceImpl` class.
Modified `Application` class and file `application.properties`.
Created a new folder `resources` in the `src/test` directory
Create a new file `application.properties` in the `src/test/resources` folder.
Added dependencies to the `pom.xml` file.

ipantazi committed:
Added `Book` entity, class `DataProcessingException`, `BookRepository` interface with methods and `BookRepositoryImpl` class.
Added `BookService` interface with methods and `BookServiceImpl` class.
Modified class `Application` and file `application.properties`.
Created a new folder `resources` in the `src/test` directory
Create a new file `application.properties` in the `src/test/resources` folder.
Added dependencies to the `pom.xml` file.

ipantazi committed:
Using EntityManagerFactory instead of SessionFactory in `BookRepositoryImpl` class

Elena-Bruyako commented on pom.xml:
```diff
@@ -54,6 +54,16 @@
+        </dependency>
+        <dependency>
+            <groupId>commons-dbcp</groupId>
+            <artifactId>commons-dbcp</artifactId>
```
Why do you need such dependency?

Elena-Bruyako commented on src/main/java/mate/academy/onlinebookstore/Application.java:
```diff
@@ -0,0 +1,33 @@
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return args -> {
+            Book macbeth = new Book();
```
```suggestion
            Book book = new Book();
```

Elena-Bruyako commented on src/main/java/mate/academy/onlinebookstore/Application.java:
```diff
@@ -0,0 +1,33 @@
+            Book macbeth = new Book();
+            macbeth.setTitle("Macbeth");
+            macbeth.setAuthor("William Shakespeare");
+            macbeth.setIsbn("978-0091547714");
```
set up all fields

Elena-Bruyako commented on src/main/java/mate/academy/onlinebookstore/model/Book.java:
```diff
@@ -0,0 +1,24 @@
+import lombok.Data;
+
+@Entity
+@Data
```
Don't use `@Data` for entities, find and read articles about problems
Lombok has a lot of other annotations that replace boilerplate code
Don't forget to add Table annotation, to specify the name of table according to the naming convention

Elena-Bruyako commented on src/main/java/mate/academy/onlinebookstore/model/Book.java:
```diff
@@ -0,0 +1,24 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
```
add required constraints for fields

Elena-Bruyako commented on src/main/java/mate/academy/onlinebookstore/repository/BookRepository.java:
```diff
@@ -0,0 +1,11 @@
+    Book save(Book book);
+
+    List<Book> findAll();
+
```
```suggestion
```

Elena-Bruyako commented on src/main/java/mate/academy/onlinebookstore/repository/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,53 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
```suggestion
@Repository
@RequiredArgsConstructor
```

Elena-Bruyako commented on src/main/java/mate/academy/onlinebookstore/repository/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,53 @@
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final EntityManagerFactory entityManagerFactory;
+
+    @Autowired
+    public BookRepositoryImpl(EntityManagerFactory entityManagerFactory) {
+        this.entityManagerFactory = entityManagerFactory;
+    }
```
```suggestion
```

Elena-Bruyako commented on src/main/java/mate/academy/onlinebookstore/service/BookServiceImpl.java:
```diff
@@ -0,0 +1,27 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
```suggestion
@Service
@RequiredArgsConstructor
```

Elena-Bruyako commented on src/main/java/mate/academy/onlinebookstore/service/BookServiceImpl.java:
```diff
@@ -0,0 +1,27 @@
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
```suggestion
```

Elena-Bruyako commented on src/main/resources/application.properties:
```diff
@@ -1 +1,9 @@
 spring.application.name=Online-Book-store
+
+spring.datasource.url=jdbc:mysql://localhost:3306/my_db?serverTimezone=UTC
```
it's better to rename schema according to your project (don't be so abstract)

Elena-Bruyako reviewed (changes requested):
see comments

ipantazi committed:
Deleted dependency `commons-dbcp` in `pom.xml`.
Set upped all fields and renamed object in class `Application`.
 Renamed schema in `application.properties`.
 Added annotation `+@RequiredArgsConstructor` and deleted constructor in classes `BookServiceImpl` and `BookRepositoryImpl`.
 Added required constraints for fields, changed annotation `@Data` to `@Getter`, `@Setter`, `@Builder`, `@ToString`, `@NoArgsConstructor`, `@AllArgsConstructor`, added annotation `@Table` and methods `equals` and `hashCode` in class `Book`.

Elena-Bruyako commented on src/main/java/mate/academy/onlinebookstore/model/Book.java:
```diff
@@ -0,0 +1,79 @@
+    private String coverImage;
+
+    @Override
+    public final boolean equals(Object o) {
```
remove

Elena-Bruyako commented on src/main/java/mate/academy/onlinebookstore/model/Book.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    @Override
+    public int hashCode() {
```
remove

Elena-Bruyako commented on src/main/java/mate/academy/onlinebookstore/model/Book.java:
```diff
@@ -0,0 +1,79 @@
+@Getter
+@Setter
+@Builder
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
```
```suggestion
```

Elena-Bruyako commented on src/main/java/mate/academy/onlinebookstore/model/Book.java:
```diff
@@ -0,0 +1,79 @@
+@Table(name = "books")
+@Getter
+@Setter
+@Builder
```
do you use builder somewhere? If no - remove

Elena-Bruyako commented on src/main/java/mate/academy/onlinebookstore/model/Book.java:
```diff
@@ -0,0 +1,79 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "title", nullable = false)
```
```suggestion
    @Column(nullable = false)
```

for all fields

Elena-Bruyako reviewed (changes requested):
see comments

ipantazi committed:
Removed methods: `equals()`, hashCode();
annotations: `@ToString`, `@NoArgsConstructor`, `@AllArgsConstructor`, `@Builder`;
 modified annotations `@Column` for all fields in `Book` class.

ahoienko reviewed (approved):
Good job!
